### ISSUE TITLE
opencl

### ISSUE THREAD

**Issue by tdi1990** [brichard19/BitCrack#245]
[2020-05-28.23:01:31] [Info] Compression: compressed
[2020-05-28.23:01:31] [Info] Starting at: 0000000000000000000000000000000000000000000000000000000000000001
[2020-05-28.23:01:31] [Info] Ending at:   FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364140
[2020-05-28.23:01:31] [Info] Counting by: 0000000000000000000000000000000000000000000000000000000000000001
[2020-05-28.23:01:32] [Info] Compiling OpenCL kernels...

and stopped help ?

**Comment by ghost** [comment]
this is problem in the invModP256k function

**Comment by tdi1990** [comment]
how to solve this problem? Thanks

**Comment by ghost** [comment]
this function needs to be rewritten in order for the kernel to compile faster. The kernel requires large optimizations. 
Just open the task manager and you will see that the kernel is compiling, you just have to wait

**Comment by tdi1990** [comment]
I waited all night ;)

**Comment by ghost** [comment]
did you build the program from source code?

**Comment by tdi1990** [comment]
i didn't understand the question well;(

**Comment by ghost** [comment]
what is your operating system?
if windows try run this file
[CLUnitTests.zip](https://github.com/brichard19/BitCrack/files/4700799/CLUnitTests.zip)

**Comment by tdi1990** [comment]
windows
and what should be?

**Comment by ghost** [comment]
the test should pass successfully

**Comment by tdi1990** [comment]
Testing device and stopped

**Comment by ghost** [comment]
Are you using the latest graphics card drivers?

**Comment by tdi1990** [comment]
yes

**Comment by ghost** [comment]
Try the optimized version
[CLUnitTests.zip](https://github.com/brichard19/BitCrack/files/4701865/CLUnitTests.zip)

**Comment by tdi1990** [comment]
Found 1 devices
Testing device          Intel(R) Pentium(R) CPU B940 @ 2.00GHz
and stop

**Comment by ghost** [comment]
I think the problem is in your processor. The program should find 2 devices, a video card and a processor. Your video card either does not support opencl or you have something with the drivers. What is your graphics card?

**Comment by tdi1990** [comment]
intel(R) HD graphics 
problem here ?

**Comment by ghost** [comment]
> problem here ?

Yes. Considering the processor, you have a built-in video card and it does not support opencl

**Comment by tdi1990** [comment]
that's what to do ? :(
